connectors/citrus-openapi/src/main/java/org/citrusframework/openapi/model/v3/Oas30ModelHelper.java

```diff
@@ -172,44 +175,69 @@ public static Optional<String> getRequestContentType(Oas30Operation operation) {
+                .filter(entry -> entry.getValue().schema != null)
+                .map(Map.Entry::getKey)
+                .findFirst());
+    }
 
-        for (OasResponse response : operation.responses.getResponses()) {
-            if (response.$ref != null) {
-                responses.add(openApiDoc.components.responses.get(OasModelHelper.getReferenceName(response.$ref)));
-            } else {
-                responses.add(response);
-            }
+    public static Optional<OasResponse> getResponseForRandomGeneration(Oas30Document openApiDoc, Oas30Operation operation) {
+        if (operation.responses == null) {
+            return Optional.empty();
         }
 
+        List<OasResponse> responses = OasModelHelper.resolveResponses(operation.responses,
+            responseRef -> openApiDoc.components.responses.get(OasModelHelper.getReferenceName(responseRef)));
+
         // Pick the response object related to the first 2xx return code found
-        Optional<Oas30Response> response = responses.stream()
-                .filter(Oas30Response.class::isInstance)
-                .filter(r -> r.getStatusCode() != null && r.getStatusCode().startsWith("2"))
-                .map(Oas30Response.class::cast)
-                .filter(res -> Oas30ModelHelper.getSchema(res).isPresent())
-                .findFirst();
+        Optional<OasResponse> response = responses.stream()
+            .filter(Oas30Response.class::isInstance)
+            .filter(r -> r.getStatusCode() != null && r.getStatusCode().startsWith("2"))
+            .map(OasResponse.class::cast)
+            .filter(res -> OasModelHelper.getSchema(res).isPresent())
+            .findFirst();
 
         // No 2xx response given so pick the first one no matter what status code
-        if (!response.isPresent()) {
+        if (response.isEmpty()) {
+            // TODO: This behavior differs from OAS2 and is very likely a bug because it may result in returning error messages.
+            // According to the specification, there MUST be at least one response, which SHOULD be a successful response.
+            // If the response is NOT A SUCCESSFUL one, we encounter an error case, which is likely not the intended behavior.
+            // The specification likely does not intend to define operations that always fail. On the other hand, it is not
+            // against the spec to NOT document an OK response that is empty.
+            // For testing purposes, note that the difference between OAS2 and OAS3 is evident in the Petstore API.
+            // The Petstore API specifies successful response codes for OAS3 but lacks these definitions for OAS2.
+            // Therefore, while tests pass for OAS3, they fail for OAS2.
+            // I would suggest to return an empty response in case we fail to resolve a good response, as in Oas2.
+            // In case of absence of a response an OK response will be sent as default.
```

Comment:
> I would suggest to return an empty response in case we fail to resolve a good response, as in Oas2.

so, if any response is defined, but no OK response (200 <= code < 300) - you would still return an empty OK response?

> In case of absence of a response an OK response will be sent as default.

in absence of **any** response, an OK response can be constructed. I am ok with that.